src/events.js

```diff
@@ -192,29 +192,35 @@ export const addEventAddSpecimenToBox = () => {
-            return;
-        }
-        
-        if(isScannedIdInUnshippedBoxes) {
-            const boxNum = scannedIdInUnshippedBoxes[conceptIds.shippingBoxId];
-            const siteSpecificLocation = conceptIdToSiteSpecificLocation[scannedIdInUnshippedBoxes[conceptIds.shippingLocation]];
-            const siteSpecificLocationName = siteSpecificLocation || '';
-            const scannedInput = scannedIdInUnshippedBoxes['inputScanned'];
-            showNotifications({ title:`${scannedInput} has already been recorded`, body: `${scannedInput} is recorded as being in ${boxNum} in ${siteSpecificLocationName}`});
-            return;
+
+        // Find the masterSpecimenId in available collections. Else, show a notification and return early.
+        const canAddSpecimen = searchAvailableCollectionsForSpecimen(masterSpecimenId);
+
+        if (!canAddSpecimen) {
+            // Check if the scanned id is already in an unshipped box. If not, check if the scanned id is already shipped. Show a notification and return early.
+            const scannedIdInUnshippedBoxes = findScannedIdInUnshippedBoxes(allBoxesList, masterSpecimenId);
+            const isScannedIdInUnshippedBoxes = scannedIdInUnshippedBoxes['foundMatch'];
+            if(isScannedIdInUnshippedBoxes) {
+                const boxNum = scannedIdInUnshippedBoxes[conceptIds.shippingBoxId];
+                const siteSpecificLocation = conceptIdToSiteSpecificLocation[scannedIdInUnshippedBoxes[conceptIds.shippingLocation]];
+                const siteSpecificLocationName = siteSpecificLocation || '';
+                const scannedInput = scannedIdInUnshippedBoxes['inputScanned'];
+                showNotifications({ title:`${scannedInput} has already been recorded`, body: `${scannedInput} is recorded as being in ${boxNum} in ${siteSpecificLocationName}`});
+                return;
+            } else {
+                const foundScannedIdShipped = await isScannedIdShipped(allBoxesList, masterSpecimenId);
+                if (foundScannedIdShipped){
+                    showNotifications({ title:'Item reported as already shipped', body: 'Please enter or scan another specimen bag ID or Full Specimen ID.'});
+                    return;
+                }
+            }
         }
 
         const specimenTablesResult = buildSpecimenDataInModal(masterSpecimenId);
         const biospecimensList = specimenTablesResult.biospecimensList;
 
         if (biospecimensList.length === 0) {
-            showNotifications({ title: 'Item not found', body: `Item not reported as collected. Go to the Collection Dashboard to add specimen.` });
+            showNotifications({ title: 'Item not found', body: `Item not reported as collected or item is in an existing available collection. If item doesn't have a matching Specimen Bag ID in available collections, go to the Collection Dashboard to add specimen.` });
             return;
```

Comment:
I think this `return` keeps us from calling `addEventSubmitSpecimenBuildModal()` a few lines below it.

Reply:
looks like `addEventSubmitSpecimenBuildModal` is outside of `form.addEventListener`

Reply:
Thanks for spelling that out, Gene. I looked closer -- you and @jhflorey are correct about this. I can remove it, or, if you think it makes sense to retain it since it matches the structure of other showNotifications() calls (readability benefit), I can leave it.